main.c

```diff
@@ -0,0 +1,464 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+#define NUM_LEVELS 12
+#define NUM_CLASSES 10
+#define MAX_NAME_LENGTH 50
+#define MAX_PHONE_LENGTH 11
+
+typedef struct student {
+    char first_name[MAX_NAME_LENGTH];
+    char last_name[MAX_NAME_LENGTH];
+    char phone[MAX_PHONE_LENGTH];
+    int grades[10];
+    struct student* next;
+} Student;
+
+Student* create_student(char* first_name, char* last_name, char* phone, int* grades) {
+    Student* new_student = (Student*)malloc(sizeof(Student));
+    if (new_student != NULL) {
+        strncpy(new_student->first_name, first_name, MAX_NAME_LENGTH - 1);
+        new_student->first_name[MAX_NAME_LENGTH-1]='\0';
+        strncpy(new_student->last_name, last_name, MAX_NAME_LENGTH - 1);
+        new_student->last_name[MAX_NAME_LENGTH-1]='\0';
+        strncpy(new_student->phone, phone, MAX_PHONE_LENGTH - 1);
+        new_student->phone[MAX_PHONE_LENGTH-1]='\0';
+        for (int i = 0; i < 10; i++) {
+            new_student->grades[i] = grades[i];
+        }
+        new_student->next = NULL;
+    }
+    return new_student;
+}
+
+void add_student_to_class(Student** class, Student* new_student) {
+    if (*class == NULL) {
+        *class = new_student;
+    } else {
+        Student* current = *class;
+        while (current->next != NULL) {
```

Comment:
If you need to insert a new student to a 1 million students linked list, will you go all the way to the end
to add the student? Think of a more efficient way.